datafusion/physical-plan/src/execution_plan.rs

```diff
@@ -430,6 +430,32 @@ pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
         Ok(Statistics::new_unknown(&self.schema()))
     }
 
+    /// Returns statistics for a specific partition of this `ExecutionPlan` node.
+    /// If statistics are not available, should return [`Statistics::new_unknown`]
+    /// (the default), not an error.
+    /// If `partition` is `None`, it returns statistics for the entire plan.
+    fn partition_statistics(&self, partition: Option<usize>) -> Result<Statistics> {
```

Comment:
Todo: mark `fn statistics(&self) -> Result<Statistics>` as deprecated

Reply:
Done in https://github.com/apache/datafusion/pull/15852/commits/b1f701f7a0f396d592159206f517b89c8596777b